nomenclature/codelist.py

```diff
@@ -339,28 +339,28 @@ def check_illegal_characters(self, config: NomenclatureConfig) -> dict[str, Code
         illegal = ["{", "}"] + config.illegal_characters
         errors = ErrorCollector()
 
-        def _check_string(value):
+        def _check_string(attr, value):
             if isinstance(value, str):
-                if any(char in value for char in illegal):
+                if found := set(illegal).intersection(value):
+                    found = "', '".join(sorted(found))
                     errors.append(
                         ValueError(
-                            f"Unexpected character in {self.name}: '{code.name}'."
+                            f"Unexpected character(s) '{found}' in {self.name}.{attr} of: '{code.name}'."
```

Comment:
Suggest the following error messages:

Suggest a more concise and more readable (less pythonic) structure of the error message
```suggestion
                            f"Illegal character(s) '{found}' in {attr} of {self.name} '{code.name}'."
```

Reply:
I support this suggestion